src/Build/Logging/FancyLogger/FancyLoggerNode.cs

```diff
@@ -4,148 +4,136 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Build.Framework;
 
 namespace Microsoft.Build.Logging.FancyLogger
 { 
-    // TODO: Maybe remove
-    public enum FancyLoggerNodeType
+    public class FancyLoggerProjectNode
     {
-        None,
-        Build,
-        Project,
-        Target,
-        Task,
-        Message,
-        Warning,
-        Error
-    }
+        private static string GetUnambiguousPath(string path)
+        {
+            return Path.GetFileName(path);
+        }
 
-    public class FancyLoggerNode
-    {
-        public string Id;
-        public Dictionary<string, FancyLoggerNode> Children = new Dictionary<string, FancyLoggerNode>();
-        public FancyLoggerNode? Parent;
+        public int Id;
+        public string ProjectPath;
+        public bool Finished;
+        // Line to display project info
         public FancyLoggerBufferLine? Line;
-        public int Depth = 0;
-        public FancyLoggerNode(string id)
+        // Targets
+        public int FinishedTargets;
+        public FancyLoggerBufferLine? CurrentTargetLine;
+        public FancyLoggerTargetNode? CurrentTargetNode;
+        // Messages, errors and warnings
+        List<Object> AdditionalDetails = new();
+        public FancyLoggerProjectNode(ProjectStartedEventArgs args)
         {
-            Id = id;
+            Id = args.ProjectId;
+            ProjectPath = args.ProjectFile!;
+            Finished = false;
+            FinishedTargets = 0;
         }
-        public FancyLoggerNode(int id, FancyLoggerNodeType type)
+
+        public void Log()
         {
-            switch (type)
+            string lineContents = ANSIBuilder.Alignment.SpaceBetween(
```

Comment:
nit:
This is a bit complicated. I think I understand what's going on, but saving off some intermediates before getting here would make my life easier 🙂

Also, you told me you tested things like that checkmark in various consoles, including older ones like cmd.exe, right? Just want to make sure they can all render it properly.

Reply:
Yes. I tested each and the whole FancyLogger is disabled for consoles that do not support ANSI. conhost (not Windows Terminal) works well 
![image](https://user-images.githubusercontent.com/5952839/209774457-a89bb5c3-8e37-4f8b-85cb-01b1d3980d27.png)